### ISSUE TITLE
Firming up the idea

### ISSUE THREAD

**Issue by betatim** [betatim/openscienceprize#9]
Build the infrastructure required to create and publish scientific output that is more than a simple, static document. To make this a success two things are needed:

This openscienceprize project will create a web app that allows you to display a single, interactive notebook for your publication. (Build on top of experience from  [`thebe`](https://github.com/oreillymedia/thebe)). Use ORCID ID as auth. Build in the components of the "social web". Meaning ability to :star: and fork publications. (or alternative ways to have "the crowd" assign credit/fame/prestige)

In order for people to be able to publish in such a place they need to know how to create a publication that is reusable. The technical components required to build a data analysis that is geared towards reusability and richer publication format already exist (github + docker + jupyter + snakemake). To use these tools right now, you have to be more of a geek than your average researcher is. What is missing is the social component. This openscienceprize project will create (and contribute to existing training initiatives #6) a blue print for setting up such a environment.

**Comment by betatim** [comment]
usability wise make it as easy as using the arxiv or even partner with them. Upload individual files, a tar.gz, or link to a git repo. Normal users wouldn't have to figure out `git` @takluyver suggested running static analysis on the repo to figure out missing files, then execute the notebook to find missing data files. Tool for finding dependencies: https://github.com/ericdill/depfinder

**Comment by tritemio** [comment]
It would be cool to provide a web service to easily create an environment containing Jupyter Notebook, execute it remotely (a la mybinder.org) and then provide some way to download the container and easily run it locally.

Running locally allows, among other things,  to quickly process new datatsets (I'm thinking about medium data, ~1GB) without the need to upload them first.

**Comment by betatim** [comment]
You mean as a way to get going with a new analysis?

I was thinking we should create a tool that allows you to do exactly this locally, something like:
```
mkdir new-project
cd new-project
openscience init
```
which creates all the things necessary to run locally.

Should add `openscience fork <something-something>` as a way to quickly get the environment of paper `something-something` locally.

**Comment by tritemio** [comment]
@betatim, yes. I was thinking of a way to get people hooked. If you try it online and you like it, you can download/clone it and run it locally. Locally you can more easily develop the analysis/paper. Also, as a user, you want to be in control of your work. So even if you have a copy online that is runnable, you want all the data locally and be able to run it on your own computer.

Orthogonally, the idea of a "fork" command to duplicate local environments is really cool.

**Comment by tritemio** [comment]
Following the concept of a GitHub for science, a useful tool would be a github desktop/SourceTree for science projects. A GUI that lets you create a new repo with a default structure, initializes the environment (for example with conda), and gives you a "button" to activate the environment and launch a notebook.

Another "button" could push the repo to github an automatically set it up for execution on mybinder.

**Comment by cranmer** [comment]
An extension of this idea is that I've wanted to setup a docker image, launch it in the cloud somewhere, and then point `thebe` to it. Currently I can make the image, but I stopped when I realized I needed to launch my own `tempnb` server. Ideally I could have a generic `tempnb` server where I get to specify my docker image in the `thebe` javascript.

**Comment by betatim** [comment]
For that you want to watch jupyter/enhancement-proposals#12

**Comment by betatim** [comment]
Thanks for the pointer! The discussion was in jupyter/nbviewer#557
